internal/backends/ent/schema/annotation.go

```diff
@@ -25,18 +26,33 @@ func (Annotation) Mixin() []ent.Mixin {
 
 func (Annotation) Fields() []ent.Field {
 	return []ent.Field{
+		field.String("node_id").Optional().Nillable().NotEmpty(),
 		field.String("name"),
 		field.String("value"),
 		field.Bool("is_unique").
 			Default(false),
 	}
 }
 
+func (Annotation) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("node", Node.Type).
+			Ref("annotations").
+			Unique().
+			Field("node_id"),
+	}
+}
+
 func (Annotation) Indexes() []ent.Index {
 	return []ent.Index{
+		index.Fields("node_id", "name", "value").
+			Unique().
+			Annotations(entsql.IndexWhere("node_id IS NOT NULL AND TRIM(node_id) != ''")).
+			StorageKey("idx_node_annotations"),
 		index.Fields("document_id", "name", "value").
 			Unique().
-			StorageKey("idx_annotations"),
+			Annotations(entsql.IndexWhere("document_id IS NOT NULL AND TRIM(document_id) != ''")).
+			StorageKey("idx_document_annotations"),
 		index.Fields("document_id", "name").
 			Unique().
 			Annotations(entsql.IndexWhere("is_unique = true")).
```

Comment:
Use just `is_unique` for the clause

```suggestion
			Annotations(entsql.IndexWhere("is_unique")).
```

On a separate note, do you think it should additionally have the same conditions as the regular index clause? i.e.

```go
Annotations(entsql.IndexWhere("document_id IS NOT NULL AND TRIM(document_id) != '' AND is_unique")).
```